### PR TITLE
test(e2e,playwright): progress-bar.tsx

### PR DESCRIPTION
**Pull request by Aman1919**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Related to https://github.com/freeCodeCamp/freeCodeCamp/issues/51705

### CODE REVIEW

File: e2e/progress-bar.spec.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { test, expect } from '@playwright/test';
+
+test.describe('Should show the progress bar showing the completed percent on  challenges', () => {
+  test('Should enter code in the editor and show the progress bar', async ({
+    page
+  }) => {
+    test.slow();
```
Comment (naomi-lgbt) [inline e2e/progress-bar.spec.ts:7]: We shouldn't need to bump the timeout for this one.

Comment (Aman1919) [reply e2e/progress-bar.spec.ts]: removed it .

**Comment by Aman1919** [comment]
All the checks were successful.

File: e2e/progress-bar.spec.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+test.describe('Should show the progress bar showing the completed percent on  challenges', () => {
+  test('Should enter code in the editor and show the progress bar', async () => {
+    const monacoEditor = page.locator(
+      '.react-monaco-editor-container textarea'
+    );
+    await monacoEditor.click();
+
+    await page.keyboard.press('Control+A');
+    await page.keyboard.press('Meta+A');
+    await page.keyboard.press('Backspace');
```
[the remark this answers is not in the document]
Comment (Aman1919) [reply e2e/progress-bar.spec.ts:25]: Control + A wasn't working in webkit . I checked playwright documentation  . It says to use Meta + A  in Mac

**Comment by Aman1919** [comment]
did the suggested changes

File: e2e/progress-bar.spec.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    await monacoEditor.click();
+    await page.keyboard.press('Control+A');
+    //Meta + A works in webkit
+    await page.keyboard.press('Meta+A');
+    await page.keyboard.press('Backspace');
+
+    await page.keyboard.insertText(
+      '<html><body><h1>CatPhotoApp</h1><h2>Cat Photos</h2></body></html>'
+    );
+    const lowerCheckButton = page.getByText('Check Your Code (Ctrl + Enter)');
```
Comment (Ahmadkashif) [inline e2e/progress-bar.spec.ts:28]:
```suggestion
    const lowerCheckButton = page.getByRole('button', { name: 'Check Your Code (Ctrl + Enter)' }).click();
```

File: e2e/progress-bar.spec.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    await lowerCheckButton.click({ force: true });
```
Comment (Ahmadkashif) [inline e2e/progress-bar.spec.ts:30]:
```suggestion
```

File: e2e/progress-bar.spec.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    const progressBarContainer = page.getByTestId('progress-bar-container');
+    await expect(progressBarContainer).toContainText(
+      'Learn HTML by Building a Cat Photo App'
+    );
+    await expect(progressBarContainer).toContainText('0% complete');
+    const submitCheckButton = page.getByText('Submit and go to next challenge');
```
Comment (Ahmadkashif) [inline e2e/progress-bar.spec.ts:36]:
```suggestion
    const submitCheckButton = page.getByRole('button', { name: 'Submit and go to next challenge' }).click();
```

File: e2e/progress-bar.spec.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    await expect(submitCheckButton).toBeVisible();
```
Comment (Ahmadkashif) [inline e2e/progress-bar.spec.ts:37]:
```suggestion
```

**Review by Ahmadkashif — CHANGES_REQUESTED** [review]
Hi @Aman1919 👋 

This is a solid attempt from your side. I have just added a few suggestions that should make the PR golden.

If we apply these changes, the PR gets one step closer to getting merged.

**Pull request closed without merging**